components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/definitions/GraphQLSchemaDefinition.java

```diff
@@ -197,4 +214,90 @@ protected void addQueryParams(JSONObject pathsObject) {
+        String apiProviderName = api.getId().getProviderName();
+        String resourcePath = APIUtil.getGraphqlDefinitionFilePath(apiName, apiVersion, apiProviderName);
+        try {
+            String saveResourcePath = resourcePath + apiProviderName + APIConstants.GRAPHQL_SCHEMA_PROVIDER_SEPERATOR +
+                    apiName + apiVersion + APIConstants.GRAPHQL_SCHEMA_FILE_EXTENSION;
+            Resource resource;
+            if (!registry.resourceExists(saveResourcePath)) {
+                resource = registry.newResource();
+            } else {
+                resource = registry.get(saveResourcePath);
+            }
+
+            resource.setContent(schemaDefinition);
+            resource.setMediaType(String.valueOf(ContentType.TEXT_PLAIN));
+            registry.put(saveResourcePath, resource);
+            if (log.isDebugEnabled()) {
+                log.debug("Successfully imported the schema: " + schemaDefinition );
+            }
+
+            String[] visibleRoles = null;
+            if (api.getVisibleRoles() != null) {
+                visibleRoles = api.getVisibleRoles().split(",");
+            }
+
+            //Need to set anonymous if the visibility is public
+            APIUtil.clearResourcePermissions(resourcePath, api.getId(), ((UserRegistry) registry).getTenantId());
+            APIUtil.setResourcePermissions(apiProviderName, api.getVisibility(), visibleRoles, resourcePath);
+
+        } catch (RegistryException e) {
+            String errorMessage = "Error while adding Graphql Definition for " + apiName + '-' + apiVersion;
+            log.error(errorMessage, e);
+            handleException(errorMessage, e);
+        }
+    }
+
+    /**
+     * Returns the graphQL content in registry specified by the wsdl name
+     *
+     * @param apiId Api Identifier
+     * @return graphQL content matching name if exist else null
+     */
+    public String getGraphqlSchemaDefinition(APIIdentifier apiId, Registry registry) throws APIManagementException {
+        String apiName = apiId.getApiName();
+        String apiVersion = apiId.getVersion();
+        String apiProviderName = apiId.getProviderName();
+        String resourcePath = APIUtil.getGraphqlDefinitionFilePath(apiName, apiVersion, apiProviderName);
+
+        String schemaDoc = null;
+        String schemaName = apiId.getProviderName() + APIConstants.GRAPHQL_SCHEMA_PROVIDER_SEPERATOR +
+                apiId.getApiName() + apiId.getVersion() + APIConstants.GRAPHQL_SCHEMA_FILE_EXTENSION;
+        String schemaResourePath = resourcePath + schemaName;
+        try {
+            if (registry.resourceExists(schemaResourePath)) {
+                Resource schemaResource = registry.get(schemaResourePath);
+                schemaDoc = IOUtils.toString(schemaResource.getContentStream(),
+                        RegistryConstants.DEFAULT_CHARSET_ENCODING);
+            }
+        } catch (org.wso2.carbon.registry.core.exceptions.RegistryException e) {
```

Comment:
Shall we move import statement to the top of file rather than doing it inline?

Reply:
There is an another import as rg.wso2.carbon.registry.api.RegistryException at the top. I think here also we can use it instead of org.wso2.carbon.registry.core.exceptions.RegistryException